modules/candidate_list/jsx/candidateListIndex.js

```diff
@@ -64,7 +64,17 @@ class CandidateListIndex extends Component {
   fetchData() {
     return fetch(this.props.dataURL, {credentials: 'same-origin'})
       .then((resp) => resp.json())
-      .then((data) => this.setState({data}))
+      .then((data) => {
+        // Convert concatenated string of subproject and visit labels to array
+        data.Data = data.Data.map((row) => {
```

Comment:
could you do all this in `php/candidate_list.class.inc`? manipulate result of `\Utility::getVisitList()` so you get an associative array i.e. 'V1'=>'V1', 'V2'=>'V2'

i personally don't think this should be handled on the client side